test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/NullRestrictedTypeOptTests.java

```diff
@@ -69,46 +82,60 @@ static public void testEAOpportunity1() throws Throwable {
 		}
 	}
 
+	@NullRestricted
+	static PrimPair evalTestEA1a_p1;
```

Comment:
Is there a reason to change local variables p1 to p4 to static variables ? Defining them as static might change what these `evalTestEAXX` tests are originally designed for (exercising escape analysis).

Reply:
The annotation is not defined for local variables.

Reply:
If the test case is not testing null, whether the variable is null restricted or not does not matter. We can just use local variable like `PrimPair p1`

Reply:
> Defining them as static might change what these evalTestEAXX tests are originally designed for (exercising escape analysis). 

I have the same concern. If they are not local variables, EA might not be triggered. With the latest spec change, I'm not too sure if all these `evalTestEA* `tests would still behave as expected.  I'll let @hzongaro comment further

Reply:
I agree with @hangshao0 and @a7ehuo here - the point of the test was to exercise Escape Analysis with null-restricted types.  By assigning the value to a static field, it automatically escapes, so there's no longer an opportunity for stack allocation.

I would suggest simply disabling the test for now, just in case null-restricted types ever become first class types.

Reply:
Sorry - let me withdraw that suggestion.  I think it would be OK to keep the test active, and instead change the static fields back to local variables.  Even if the local variables cannot be null-restricted, there are still opportunities for Escape Analysis to perform stack allocation.

Reply:
Thanks Henry, I will update the tests with your suggestion. Once javac is updated and we transition to the tests in src_lw5 the fields can be marked as null restricted https://github.com/eclipse-openj9/openj9/blob/master/test/functional/Valhalla/src_lw5/org/openj9/test/lworld/NullRestrictedTypeOptTests.java

Reply:
Sorry.  When I was reviewing this, I didn't realize that this was for the src_qtypes version of the test rather than the src_lw5 version of the test.  I think your updates look good.

Reply:
LGTM. Thanks for the update!